File: nearly_lucky_number.cpp

#include<iostream>
using namespace std;
 int main()
 {
      long long n;
       cin>>n;
       int c=0;
       while(n>0)
       {
         if(n%10==7 || n%10==4) c++;
         n/=10;
       }
       if(c==7 || c==4) cout<<"YES";
       else cout<<"NO";
 }

File: nearly_lucky_number.java
import java.math.BigInteger;
import java.util.Scanner;
public class nearly_lucky_number {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
         BigInteger n=sc.nextBigInteger();
         String str=BigInteger.toString(n);
        if(lucky==7 || lucky==4) System.out.println("YES");
        else System.out.println("NO");
    }
    
}
